src/agent.spec.js

```diff
@@ -1,50 +1,123 @@
+    some: 'event',
+  };
+
   beforeAll(() => {
     handleTransaction = provideHandleTransaction([
-      mockAdminEventsAgent,
-      mockDeployerWatchAgent,
-      mockMonitorMintCallsAgent,
+      txHandler1,
+      txHandler2,
+      txHandler3,
+    ]);
+    handleBlock = provideHandleBlock([
+      blockHandler1,
+      blockHandler2,
+      blockHandler3,
+    ]);
+    initialize = provideInitialize([
+      txHandler1,
+      txHandler2,
+      txHandler3,
+      blockHandler1,
+      blockHandler2,
+      blockHandler3,
     ]);
   });
 
   describe('handleTransaction', () => {
-    it('invokes admin-events, deployer-watch and monitor-mint-calls agents and returns their findings', async () => {
+    it('invokes transaction handlers 1 through 3 and returns their findings', async () => {
       const mockFinding = { some: 'finding' };
-      mockAdminEventsAgent.handleTransaction.mockReturnValueOnce([mockFinding]);
-      mockDeployerWatchAgent.handleTransaction.mockReturnValueOnce([mockFinding]);
-      mockMonitorMintCallsAgent.handleTransaction.mockReturnValueOnce([mockFinding]);
+      txHandler1.handleTransaction.mockReturnValueOnce([mockFinding, mockFinding]);
+      txHandler2.handleTransaction.mockReturnValueOnce([mockFinding]);
+      txHandler3.handleTransaction.mockReturnValueOnce([]);
 
       const findings = await handleTransaction(mockTxEvent);
 
       expect(findings).toStrictEqual([mockFinding, mockFinding, mockFinding]);
-      expect(mockAdminEventsAgent.handleTransaction).toHaveBeenCalledTimes(1);
-      expect(mockAdminEventsAgent.handleTransaction).toHaveBeenCalledWith(
+      expect(txHandler1.handleTransaction).toHaveBeenCalledTimes(1);
+      expect(txHandler1.handleTransaction).toHaveBeenCalledWith(
         mockTxEvent,
       );
-      expect(mockDeployerWatchAgent.handleTransaction).toHaveBeenCalledTimes(1);
-      expect(mockDeployerWatchAgent.handleTransaction).toHaveBeenCalledWith(
+
+      expect(txHandler2.handleTransaction).toHaveBeenCalledTimes(1);
+      expect(txHandler2.handleTransaction).toHaveBeenCalledWith(
         mockTxEvent,
       );
-      expect(mockMonitorMintCallsAgent.handleTransaction).toHaveBeenCalledTimes(1);
-      expect(mockMonitorMintCallsAgent.handleTransaction).toHaveBeenCalledWith(
+
+      expect(txHandler3.handleTransaction).toHaveBeenCalledTimes(1);
+      expect(txHandler3.handleTransaction).toHaveBeenCalledWith(
         mockTxEvent,
       );
     });
   });
+
+  describe('handleBlock', () => {
+    it('invokes block handlers 1 through 3 and returns their findings', async () => {
+      const mockFinding = { some: 'finding' };
+      blockHandler1.handleBlock.mockReturnValueOnce([mockFinding, mockFinding]);
+      blockHandler2.handleBlock.mockReturnValueOnce([mockFinding]);
+      blockHandler3.handleBlock.mockReturnValueOnce([]);
+
+      const findings = await handleBlock(mockTxEvent);
+
+      expect(findings).toStrictEqual([mockFinding, mockFinding, mockFinding]);
```

Comment:
yeah, if I were to do that I would have to use `set()` since we really aren't trying to test order. I feel like this is a fine solution as is for now. I was thinking this exact thing when I was going through the findings